doc/CMakeLists.txt

```diff
@@ -66,39 +66,22 @@ else ()
   set( OUT_OF_TREE_BUILD "True" )
 endif ()
 
-# Determine doc paths
-set( DOC_BUILD_DIR "${PROJECT_BINARY_DIR}/doc/htmldoc" )
-set( DOC_SOURCE_DIR "${PROJECT_SOURCE_DIR}/doc/htmldoc" )
 
 if ( BUILD_SPHINX_DOCS )
-  # Configure the Sphinx documentation.
-  # 1. In out-of-tree builds, we need to copy over the source doc dir to the build dir,
-  #    exact reasons currently unknown.
-  # 2. The NEST source directory is passed to Sphinx by setting env:NESTSRCDIR, so that
-  #    Sphinx can rewrite the source `pynestkernel.pyx` into a mocked `pynestkernel.py`
-  # 3. Sphinx builds the documentation into the build directory, which then later gets
-  #    installed into the target docdir.
-  # 4. Sphinx generates the models RST documentation from the models/*.h files. But we
-  #    need to copy them into a deeper directory.
-  # 5. To make Python's `help` results more useful, `include` directives are inlined using
-  #    `resolve_includes.py`
-  # 6. This assembled Sphinx folder is marked to be installed, together with the logos.
+  message( STATUS "Configuring Sphinx documentation" )
+  set( _SPHINX_SOURCE_DIR "${PROJECT_SOURCE_DIR}/doc/htmldoc" )
+  set( _SPHINX_BUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}/_build/html" )
   add_custom_target( sphinxdocs
```

Comment:
When it comes generating the documentation from Sphinx, although it mainly is userdocs, it does also contain docs relevant and useful only to developers. There is a whole section on contributing to nest code and docs, and fundamentally that is not necessary user knowledge (this is I think a gray area: at what point does a user become a developer?). It also should, in the future, be home to the actual doxygen docs.  

So I know that Sphinx is merely the tool, but it is what we are using now.  The term htmldoc was chosen to move away from explicit use of user because the developer_space/ is within that directory. And it describes the role of this directory: to provide the files for documentation on the web.

Reply:
My 2 cents is that I opted for this separation so that there are 2 layers: userdoc/devdoc as options, toggling and configuring indeed whichever tool we use to build things,  and this doesn't have to be a 1 to 1 relationships, perhaps in the future multiple builds will form the user doc together. Naming it sphinx docs also makes sense as you'll see that anything inside of the block relates strictly to sphinx commands and would have to be replaced in its entirety if we switch to another tool

Reply:
Alright. Thanks for explaining again. I try to remain silent about this from now on ;-)